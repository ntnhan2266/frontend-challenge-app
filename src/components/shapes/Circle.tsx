import React, { RefObject } from "react";
import "components/shapes/circle.css";
import { SHAPE_TYPE } from "utils/shape";
import withDragDrop from "components/hocs/withDragDrop";
import { ShapeComponentType } from "common/type";

export interface CircleProps {
  cx: number;
  cy: number;
  r: number;
}

interface Props extends CircleProps {
  onDrag: Function;
  isIntersected: boolean;
  domNodeRef: RefObject<any>;
}

export const Circle: React.FunctionComponent<Props> = (props: Props) => {
  return (
    <React.Fragment>
      <circle
        className={props.isIntersected ? "fill-red" : "fill-green"}
        ref={props.domNodeRef}
        cx={props.cx}
        cy={props.cy}
        r={props.r}
      />
    </React.Fragment>
  );
};

const CircleComponent: ShapeComponentType<any> = Object.assign(Circle, {
  ShapeType: SHAPE_TYPE.CIRCLE,
});
export default withDragDrop(CircleComponent);
