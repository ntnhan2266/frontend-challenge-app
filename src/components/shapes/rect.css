.fill-green {
  fill: green;
}

.fill-red {
  fill: red;
}

.rect {
  stroke: black;
  fill: white;
  stroke-opacity: 1;
  stroke-width: 1px;
}
