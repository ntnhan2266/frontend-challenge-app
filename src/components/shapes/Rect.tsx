import React from "react";
import "components/shapes/rect.css";

export interface RectProps {
  x: number;
  y: number;
  width: number;
  height: number;
}

interface Props extends RectProps {}

export const Rect: React.FunctionComponent<Props> = (props: Props) => {
  return (
    <React.Fragment>
      <rect
        className="rect"
        x={props.x}
        y={props.y}
        width={props.width}
        height={props.height}
      />
    </React.Fragment>
  );
};

export default Rect;
