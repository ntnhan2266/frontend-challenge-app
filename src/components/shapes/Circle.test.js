import React from "react";
import { shallow } from "enzyme";
import { Circle } from "components/shapes/Circle";
let wrapped = shallow(<Circle />);

describe("Circle", () => {
  it("should render the Circle Component correctly", () => {
    expect(wrapped).toMatchSnapshot();
  });
});
