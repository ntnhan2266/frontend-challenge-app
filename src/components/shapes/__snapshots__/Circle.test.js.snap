// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle should render the Circle Component correctly 1`] = `
<Fragment>
  <circle
    className="fill-green"
  />
</Fragment>
`;
